flow/project.py

```diff
@@ -1692,18 +1715,86 @@ def _store_bundled(self, operations):
         else:
             h = '.'.join(op.id for op in operations)
             bid = '{}/bundle/{}'.format(self, sha1(h.encode('utf-8')).hexdigest())
-            fn_bundle = self._fn_bundle(bid)
+            fn_bundle = self._fn_stored('.bundles', bid)
             os.makedirs(os.path.dirname(fn_bundle), exist_ok=True)
             with open(fn_bundle, 'w') as file:
                 for operation in operations:
                     file.write(operation.id + '\n')
             return bid
 
+    def _store_aggregates(self, operations):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param operations:
+            The operations to be submitted.
+        :type operations:
+            A sequence of instances of :py:class:`.JobOperation`
+        """
```

Comment:
So what we're essentially storing here is:
`op.name -> [operation.id <aggregate job ids>]*`
Is that correct?

Is there a reason that we don't just store those in the project document or so?

Reply:
Yes we're storing the mentioned data only.
And no there's no specific reason to do this. I thought that a structure similar to bundles would be helpful.

Reply:
I think we should avoid adding more files to the project space if at all possible. I believe we are using the file-based approach for bundles because we need to be able to access them in a very distributed manner. That said, the project document did not exist yet at the time when bundles were implemented.

Reply:
Oh okay, that makes sense now. Will use project doc

Reply:
@csadorf I'm fine with this for now, but in the long term I think we should move away from using the project document for storing internal signac state information. I agree that avoiding polluting the user's project space is a good thing for now, but when we get around to implementing glotzerlab/signac#197 I would also like to change how we deal with aggregates (and status).

Part of the reason for this is that (unlike state points), job documents (and therefore also project documents) are files that at least some users will choose to open and possibly even edit (for instance, to modify some progress or completion indicator) without using the signac API, and I think that's OK. That being the case, I think it would also be preferable not to clutter this file with information on signac's internal operations (aggregates, bundles, status, etc). At a conceptual level, I think we should clearly separate user and internal information, and using the project document this way breaks that separation. The other part of my reasoning is that for large data spaces, storing status (and aggregate/bundle/etc information) in the project document makes it large enough to generate noticeable performance delays with opening, modifying, and resaving the file. Buffering can of course ameliorate these performance issues, but I think we would be better off avoiding them completely.

I don't think making any of the changes I've mentioned is in scope for PR, but since it's relevant here I wanted to post and gauge opinions. For now (and until we move forward on glotzerlab/signac#197) I think using the project document is fine, but if you agree then we should add this to the long-term to-do list.